### ISSUE TITLE
The `package` command does not replace `process.env.*`

### ISSUE THREAD

**Issue by rgossiaux** [sveltejs/kit#3480]
### Describe the bug

When you run `npm run package` in a SvelteKit project, `process.env` references (like `process.env.NODE_ENV`) are not statically replaced; the files that are output still contain the `process.env` references. 

I don't know if there are cases when this is desirable, but in my case it is not. My library has a few checks for `process.env.NODE_ENV === "test"` which are only internal. I have a hard time imagining a case when a package user would want anything other than `"production"` from a third-party library. 

In particular, this seems like it causes an issue with Svelte projects built with https://github.com/sveltejs/template where the packaging also does not handle `process.env`. 

### Reproduction

It's not a minimal example, but I ran into this with my component library: https://github.com/rgossiaux/svelte-headlessui

You can repro this by checking out & installing the repo, then running `npm run package`. 

A user then reported that the resulting package is broken with https://github.com/sveltejs/template using this example repo: https://github.com/jerriclynsjohn/svelte-headlessui-test-repo

### System Info

```shell
System:
    OS: macOS 12.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 7.89 GB / 64.00 GB
    Shell: 3.3.1 - /opt/homebrew/bin/fish
  Binaries:
    Node: 16.8.0 - /var/folders/rn/2kdgb0jj25q2v656j3qc_ywr0000gn/T/fnm_multishells/23841_1640400023703/bin/node
    Yarn: 1.22.17 - /opt/homebrew/bin/yarn
    npm: 7.21.0 - /var/folders/rn/2kdgb0jj25q2v656j3qc_ywr0000gn/T/fnm_multishells/23841_1640400023703/bin/npm
  Browsers:
    Chrome: 97.0.4692.99
    Safari: 15.2

```
```


### Severity

serious, but I can work around it

### Additional Information

_No response_

**Comment by dummdidumm** [comment]
There are probably arguments both in favor and against this behavior, although I lean towards replacing them. Not sure how for example node libraries do this.

**Comment by Conduitry** [comment]
I don't think we want to let support in the old template drive decisions here. I think I'm against performing this replacement automatically (replaced with what? you don't know what env value consumers will want to be using), but maybe there could be some config for additional processing done during the packaging. Maybe Svelte preprocessors already would cover this? There have been some other issues opened about adjusting what happens during packaging.

**Comment by dodiameer** [comment]
Possibly related to #2572  - It would help fix this issue because (assuming bundling would use vite or rollup) you could use [Vite's `define` option](https://vitejs.dev/config/#define) or [`@rollup/plugin-replace`](https://github.com/rollup/plugins/tree/master/packages/replace) to do the replacement, but as of right now using either of them won't work.

**Comment by rgossiaux** [comment]
> I don't think we want to let support in the old template drive decisions here. I think I'm against performing this replacement automatically (replaced with what? you don't know what env value consumers will want to be using), but maybe there could be some config for additional processing done during the packaging. Maybe Svelte preprocessors already would cover this? There have been some other issues opened about adjusting what happens during packaging.

I mention the old template just as a concrete example of how this can cause problems, but it feels "wrong" to me anyway regardless of that specific example, and least in the case when you're writing a component library.

It makes sense to me if you don't want to turn this on globally (I could imagine someone writing a package that didn't want this), but I think it would make sense too for there to be a config option. As for "with what?", using the env values that are present during packaging seems like an obvious and sane default (and would presumably allow package authors to customize the output appropriately).

**Comment by Jojoshua** [comment]
@rgossiaux did you ever find a workaround?

**Comment by rgossiaux** [comment]
Sure, I'm working around this by just manually replacing `process.env.NODE_ENV` in the `package` script in `package.json`: 

https://github.com/rgossiaux/svelte-headlessui/blob/f431f891271291c8db7c75fe00c7c19b2542a21d/package.json#L22

Pretty janky but it gets the job done for my use case.

**Comment by benmccann** [comment]
You can use [`esm-env`](https://github.com/benmccann/esm-env) rather than `process.env.*` and it will work in the browser and Node without needing any replacement. Perhaps we should have a warning when `process.env` is present notifying people about that
